### PR TITLE
ref(reprocessing-v2): Re-use event ID for reprocessed events

### PR DESCRIPTION
**Pull request by untitaker**
* [x] make sure that https://github.com/getsentry/snuba/pull/1476 lands
* [x] check attachments support again
* [x] check user feedback again
* [x] make sure all consumers can deal with this
	list from filippo

    * [x] snuba events consumer -- routes to replacers: https://github.com/getsentry/snuba/blob/6f67d7cc1ba54872e2b56c086238cb0f3446f528/snuba/datasets/events_processor_base.py#L46
    * [x] snuba errors consumer  (extends from the same code of the events consumer)
    * [x] snuba transactions consumer -- should be ok: https://github.com/getsentry/snuba/blob/6f67d7cc1ba54872e2b56c086238cb0f3446f528/snuba/datasets/transactions_processor.py#L64-L65
    * [x] snuba spans consumer -- should be ok: https://github.com/getsentry/snuba/blob/6f67d7cc1ba54872e2b56c086238cb0f3446f528/snuba/datasets/spans_processor.py#L100-L101
    * [x] snuba subscriptions consumers (events and transactions but the code is basically the same) -- ted said it's ok, can't find relevant code though
    * [x] data pipelines (Syd can give you context on where the conosumers are but I think they are here https://github.com/getsentry/super-big-consumers/tree/master/consumer) -- syd confirmed there are no exhaustive mappings of message types anyplace else
    * [x] sentry post process forwarder (*in this PR*)

### CODE REVIEW

File: src/sentry/tasks/reprocessing2.py [hunk 1]
```diff
@@ -102,8 +104,9 @@ def wait_group_reprocessed(project_id, group_id):
 )
 def delete_old_group(project_id, group_id):
     from sentry.models.group import Group
-    from sentry.group_deletion import delete_group
 
     group = Group.objects.get_from_cache(id=group_id)
 
-    delete_group(group)
+    # All the associated models (groupassignee and eventattachments) should
```
Comment (untitaker) [inline src/sentry/tasks/reprocessing2.py:110]: I think this might leak tsdb data, not entirely sure

Comment (untitaker) [reply src/sentry/tasks/reprocessing2.py:110]: I *think* this should be fine as TSDB runs over Snuba.

**Comment by untitaker** [comment]
@tkaemming please review the checklist of consumers too

File: src/sentry/tasks/reprocessing2.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/tasks/reprocessing2.py:146]: Forgot similarity here 💀 💀 💀 💀 💀 💀 💀

**Comment by tkaemming** [comment]
> @tkaemming please review the checklist of consumers too

It looks comprehensive to my knowledge. The reason you can't find anything for subscriptions is because they don't actually parse the message payload — it just extracts timing information from the message metadata to drive the scheduler.

**Comment by tkaemming** [comment]
Reposting here from chat: this generally looks okay to me (aside from whatever is happening with tests), but I feel that somebody with a bit more knowledge of reprocessing (and the broader processing pipeline at this point — a lot has changed since the last time I was in there) should also check it out.

File: src/sentry/eventstream/snuba.py [hunk 2]
```diff
@@ -217,6 +217,24 @@ def end_delete_tag(self, state):
         state["datetime"] = datetime.now(tz=pytz.utc)
         self._send(state["project_id"], "end_delete_tag", extra_data=(state,), asynchronous=False)
 
+    def tombstone_events(self, project_id, event_ids):
```
Comment (untitaker) [inline src/sentry/eventstream/snuba.py:220]: check if tombstoning before going to the next page woudl invalidate the offset while paginating through group

File: src/sentry/reprocessing2.py [hunk 3]
```diff
@@ -88,26 +80,44 @@ def reprocess_event(project_id, event_id, start_time):
...
+    node_id = _generate_unprocessed_event_node_id(project_id=project_id, event_id=event_id)
+
+    with sentry_sdk.start_span(op="reprocess_events.nodestore.get"):
+        data = nodestore.get(node_id)
 
     # Step 1: Fix up the event payload for reprocessing and put it in event
     # cache/event_processing_store
-    orig_event_id = data["event_id"]
-    set_tag(data, "original_event_id", orig_event_id)
+    event_id = data["event_id"]
```
Comment (untitaker) [inline src/sentry/reprocessing2.py:90]: order this down to after `if data is none`

File: src/sentry/reprocessing2.py [hunk 4]
```diff
@@ -232,8 +237,12 @@ def start_group_reprocessing(project_id, group_id, max_events=None, acting_user_
...
         new_group.save()
 
-        for model in GROUP_MODELS_TO_MIGRATE:
-            model.objects.filter(group_id=group_id).update(group_id=new_group.id)
+    # This migrates all models that are associated with a group but not
+    # directly with an event, i.e. everything but event attachments and user
+    # reports. Those other updates are run per-event (in
+    # post-process-forwarder) to not cause too much load on pg.
+    for model in GROUP_MODELS_TO_MIGRATE:
+        model.objects.filter(group_id=group_id).update(group_id=new_group.id)
```
Comment (jan-auer) [inline src/sentry/reprocessing2.py:245]: Wasn't the point of `DIRECT_GROUP_RELATED_MODELS` to be able to do this in an atomic transaction?

File: src/sentry/tasks/post_process.py [hunk 5]
```diff
@@ -182,6 +175,10 @@ def post_process_group(
         _capture_stats(event, is_new)
 
         if event.group_id:
+            # Patch attachments that were ingested on the standalone path.
```
Comment (untitaker) [inline src/sentry/tasks/post_process.py:178]: move after plugins

File: src/sentry/reprocessing2.py (hunk 4 above)
Comment (untitaker) [reply src/sentry/reprocessing2.py:242]: not sure why I moved this out. We'll have to see if it causes problems

**Review by jan-auer — APPROVED** [review]

**Review by tonyo — APPROVED** [review]

**Pull request merged**